### PR TITLE
added bcalm+abyss pipeline

### PR DESCRIPTION
**Pull request by rchikhi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -376,3 +376,55 @@ abyss-todot --gfa1 9_assembly.gfa >9_assembly.gfa1
...
+## BCALM+ABySS
+
+Assemble reads using BCALM for unitigs, then the rest of the ABySS pipeline from the previous example.
+
+```
+# make sure you have bcalm: https://github.com/GATB/bcalm/releases
+
+# Install the dependencies
+pip install gfapy
+brew install abyss curl pigz samtools seqtk
```
Comment (sjackman) [inline README.md]: I've added `bcalm` to `brewsci/bio`

File: README.md [hunk 2]
```diff
@@ -376,3 +376,55 @@ abyss-todot --gfa1 9_assembly.gfa >9_assembly.gfa1
...
+# Download the data
+seqtk mergepe <(curl -Ls https://github.com/rrwick/Unicycler/raw/master/sample_data/short_reads_1.fastq.gz) <(curl -Ls https://github.com/rrwick/Unicycler/raw/master/sample_data/short_reads_2.fastq.gz) | gzip >0_pe.fq.gz
+# Unitig with BCALM
+bcalm -in 0_pe.fq.gz -out 1_bcalm -kmer-size 99 -abundance-min 1
+convertToGFA.py 1_bcalm.unitigs.fa 1_unitig.gfa 99
+# convert bcalm output to gfa2
+echo -e 'H\tVN:Z:1.0' > 1_unitig.gfa1
+tail -n +2 1_unitig.gfa >> 1_unitig.gfa1
```
Comment (sjackman) [inline README.md]: I'd suggest
```sh
(printf "H\tVN:Z:1.0\n"; tail -n +2 1_unitig.gfa) >1_unitig.gfa1
```
`printf` is more portable than `echo -e`.

**Comment by sjackman** [comment]
Two small comments. Looks good to me.
Since most of this is duplicated from the previous example, do you think it'd be better to split the ABySS pipeline in two: unitigs, and the rest? Then the BCALM pipeline can say something like, "Then run Phase 2 of the ABySS pipeline`.
I'd be curious to see whether we could use the unitigger stage from Unicycler/SPAdes with the contigging stage of ABySS. It should work.

File: README.md (hunk 1 above)
Comment (rchikhi) [reply README.md]: thanks!

**Comment by rchikhi** [comment]
agreed, would be better to split. let me do that

**Comment by rchikhi** [comment]
actually it's not that straightforward because I want to use k=99.
It can become a bit bloated to write "-k$(k)" everywhere instead of -k100, what do you think?

**Comment by sjackman** [comment]
You can change the ABySS example to use `-k99` instead of 100 if you prefer. Do you prefer 99 because it's odd, and is BCALM limited to odd values of *k*?

**Comment by sjackman** [comment]
Looks good to me!

**Comment by sjackman** [comment]
The file numbering scheme made sense when it was a single monolithic pipeline, but makes less sense when it's broken up into components. It's fine for now though.

**Pull request merged**